### PR TITLE
Cherry pick: check network to k8s api server

### PR DESCRIPTION
**Pull request by bowang-666**
**Reason for Change**:

Cherry pick cse script changes to aks release branch.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/engine/virtualmachinescalesets.go [hunk 1]
```diff
@@ -743,7 +743,7 @@ func CreateAgentVMSS(cs *api.ContainerService, profile *api.AgentPoolProfile) Vi
...
+		commandExec := fmt.Sprintf("[concat('echo $(date),$(hostname); %s for i in $(seq 1 1200); do grep -Fq \"EOF\" /opt/azure/containers/provision.sh && break; if [ $i -eq 1200 ]; then exit 100; else sleep 1; fi; done; ', variables('provisionScriptParametersCommon'),%s,' API_SERVER_IP=',variables('kubernetesAPIServerIP'),' IS_VHD=%s GPU_NODE=%s SGX_NODE=%s AUDITD_ENABLED=%s /usr/bin/nohup /bin/bash -c \"/bin/bash /opt/azure/containers/provision.sh >> /var/log/azure/cluster-provision.log 2>&1%s\"')]", outBoundCmd, generateUserAssignedIdentityClientIDParameter(userAssignedIDEnabled), isVHD, nVidiaEnabled, sgxEnabled, auditDEnabled, runInBackground)
```
Comment (xuto2) [inline pkg/engine/virtualmachinescalesets.go]: ideally the new API_SERVER_IP should go to 
https://github.com/Azure/aks-engine/blob/b1284c7ac9a6f9e288b0c50f931eafd0144d8cbf/pkg/api/types.go#L2315
so it's part of the variables('provisionScriptParametersCommon'). But it's fine to address this later.

FYI @CecileRobertMichon

Comment (CecileRobertMichon) [reply pkg/engine/virtualmachinescalesets.go]: Good point thanks @xuto2 

@bowang-666 do you mind fixing in a follow up PR? Thanks!

Comment (bowang-666) [reply pkg/engine/virtualmachinescalesets.go]: I will fix in another pr.

Comment (bowang-666) [reply pkg/engine/virtualmachinescalesets.go:746]: This is moved to parameters common in pr: https://github.com/Azure/aks-engine/pull/3002

**Pull request merged**